app/controllers/tickets_controller.rb

```diff
@@ -164,76 +168,80 @@ def create
         end
       end
     else
+      using_hook = false
       @ticket = Ticket.new(ticket_params)
-      if current_user.nil? && !verify_recaptcha
-        return render 'new'
-      end
     end
 
-    send_notification_email
-
-    respond_to do |format|
-      format.html do
-
-        respond_to_html
-
+    if ::Rails.application.config.ticket_is_open_to_the_world  == false && current_user.nil? && using_hook == false
+      render :status => :forbidden, :text => "Access Denied"
+    elsif can_create_a_ticket(using_hook) && save_current_ticket
+      notify_incoming @ticket
+      
+      respond_to do |format|
+        format.json { render json: @ticket, status: :created }
+        format.html {
+          if current_user.nil?
+            flash.now[:notice] = I18n::translate(:ticket_added_public)
+            render 'create'
+          else
+            redirect_to ticket_url(@ticket), notice: I18n::translate(:ticket_added)
+          end
+        }
       end
-
-      format.json do
-
-        respond_to_json
-
+    else
+      respond_to do |format|
+        format.html {
+          @email_addresses = EmailAddress.verified.ordered
+          render 'new'
+        }
+        format.json {
+          if @ticket.nil?
+            render json: {}, status: :created  # bounce mail handled correctly
+          else
+            render json: @ticket, status: :unprocessable_entity
+          end 
+        }
       end
-
-      format.js { render }
     end
   end
 
   protected
-  def respond_to_json
-    if @ticket.nil?
-      render json: {}, status: :created  # bounce mail handled correctly
-    elsif @ticket.valid?
-      render json: @ticket, status: :created
+  def can_create_a_ticket(using_hook)
+    if @ticket.nil? || !@ticket.valid?
+      flash.now[:alert] = I18n::translate(:form_validation_error)
```

Comment:
[Here it is](https://github.com/ivaldi/brimir/pull/312/commits/edf760f8887fbdd871b857b8daa020b09f7d2743). Let me know what you think